src/Elm/Syntax/TypeAnnotation.elm

```diff
@@ -46,16 +47,11 @@ type TypeAnnotation
     = Var String
     | Type (Node ( ModuleName, String )) (List (Node TypeAnnotation))
     | Tuple (List (Node TypeAnnotation))
-    | Record RecordDefinition (Maybe (Node String))
+    | Record (List (Node RecordField))
+    | ExtensionRecord (Node String) (Node RecordField) (List (Node RecordField))
     | FunctionTypeAnnotation (Node TypeAnnotation) (Node TypeAnnotation)
 
 
-{-| A list of fields in-order of a record type annotation.
--}
-type alias RecordDefinition =
-    List (Node RecordField)
```

Comment:
I removed `RecordDefinition` as it felt like unnecessary indirection when figuring out the type signature for `Record`.